### DISCUSSION TITLE
OpenGL 3.2 in proot/chroot with virgl-android ?

### DISCUSSION THREAD

**Discussion by xDoge26** [termux/termux-packages#16508] [General]
Hi, i want to use opengl 3.2 program in proot/chroot using virgl-android
However, Qualcoom stock gles driver can't expose it [v@502](https://opengles.gpuinfo.org/displayreport.php?id=6597) 

So i had to update gles driver to [v@690](https://opengles.gpuinfo.org/displayreport.php?id=6598). Problem solved , but this method require Magisk and can break some system apps

Im thinking about some solutions that dont require root, such as :
- Virgl-android with ANGLE GLES ?
- Virgl-android with ANGLE Vulkan ?
- Virgl-android with custom [Driver](https://github.com/K11MCH1/AdrenoToolsDrivers) instead of using stock driver provided by OEM (Adreno only?)

**Comment by xDoge26** [comment]
@twaik @licy183 
Can you guys help me ?

**Reply by twaik** [reply]
As I can see you can simply build needed GLES libraries and put them into some location inside termux. You can use LD_LIBRARY_PATH trick to force virgl_test_server_android use your libraries.

**Reply by xDoge26** [reply]
Can you tell me how, im not developer

**Comment by licy183** [comment]
I've added `angle-android`. Does it fix this?
